broker/data_upload/_ingest_alerts.py

```diff
@@ -0,0 +1,158 @@
+
+    error_client = error_reporting.Client()
+    bq_client = bigquery.Client()
+    log = setup_log('data_upload')
+
+SCHEMA_DIR = FILE_DIR = Path(__file__).resolve().parent / 'schema'
+
+
+def _get_table_id(data_set, table):
+    """Return the ID for a BigQuery table
+
+    Args:
+        data_set (str): The name of the data set
+        table    (str): The name of the table
+
+    Returns:
+        The name of the specified table as a string
+    """
+
+    table_ref = bq_client.dataset(data_set).table(table)
+    return f'{table_ref.dataset_id}.{table_ref.table_id}'
+
+
+def stream_ingest(data, data_set, table):
+    """Stream ingest a Pandas DataFrame into a BigQuery table
+
+    If the table does not exist, create it.
+
+    Args:
+        data (DataFrame): Data to upload to table
+        data_set   (str): The name of the data set
+        table      (str): The name of the table
+    """
+
+    project_id = os.environ['BROKER_PROJ_ID']
+    table_id = _get_table_id(data_set, table)
+    data.to_gbq(
+        table_id,
+        project_id,
+        if_exists='append',
+        progress_bar=False
+    )
+
+
+def batch_ingest(data, data_set, table):
+    """Batch upload a Pandas DataFrame into a BigQuery table
+
+    Alert data WILL be temporarily written to disk. If the table does not
+    exist, create it.
+
+    Args:
+        data (DataFrame): Data to upload to table
+        data_set   (str): The name of the data set
+        table      (str): The name of the table
+    """
+
+    # Configure batch loading
+    job_config = bigquery.LoadJobConfig()
+    job_config.source_format = bigquery.SourceFormat.AVRO
+
+    # Get tables to store data
+    table_id = _get_table_id(data_set, table)
+
+    with NamedTemporaryFile() as source_file:
+        pdx.to_avro(source_file.name, data)
+
+        try:
+            # API request
+            log.debug('Launching batch upload job.')
+            job = bq_client.load_table_from_file(
+                source_file,
+                table_id,
+                location="US",
+                job_config=job_config,
+            )
+
+        except KeyboardInterrupt:
+            job.result()
+            raise
+
+
+def upload_to_bucket(bucket_name, source_path, destination_name):
+    """Uploads a file to a GCP storage bucket
+
+    Args:
+        bucket_name      (str): Name of the bucket to upload into
+        source_path      (str): Path of the file to upload
+        destination_name (str): Name of the file to be created
+    """
+
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blob = bucket.blob(destination_name)
+    blob.upload_from_filename(source_path)
+
+
+def get_schema(schemavsn):
+    """Return the avro schema for a given ZTF schema version
+
+    Args:
+        schemavsn (str): ZTF schema version (e.g. '3.2')
+
+    Returns:
+        The schema as a dictionary
+    """
+
+    schema_path = SCHEMA_DIR / f'ztf_{schemavsn}.json'
```

Comment:
You likely mean
`schema_path = os.path.join(SCHEMA_DIR, f'ztf_{schemavsn}.json')`

Reply:
No, I avoided `.join` on purpose. Jared has converted me to using `Pathlib` over `os.path`. I find it more intuitive and easier to read.